packages/relay-runtime/store/RelayStoreTypes.js

```diff
@@ -298,6 +298,11 @@ export type Observer<T> = {
   onNext?: ?(data: T) => void,
 };
 
+export type SingleObserver<T> = {
+  onCompleted?: ?(data: T) => void,
+  onError?: ?(error: Error) => void,
+};
```

Comment:
Should change these non-optional keys or add null checks for accessing them. I think the former is more correct: the only place the keys should be optional is where the observer is a user-supplied value. It's probably better to have a PartialObserver (w optional keys) for User-supplied values and then non-optional variant for internal use. This would eliminate a lotnof redundant null checking.